### PR TITLE
BttnOne

### PR DESCRIPTION
**Pull request by WesselBruijnes**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

**Comment by natalie-pienkowska** [comment]
Hi @WesselBruijnes! Thanks for submitting your PR. Our certification engineer, Srikanth, is currently on emergency leave. Please expect a delay with your PR. He will return January 31st, 2022. We apologize for any inconveniences.

File: certified-connectors/BttnV2/Readme.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+## BttnV2 Connector
+The magical bttn that automates your business, improves your customer experience, and so much more…   
+This connector is for the bttn ONE. The next generation of Internet of Things buttons.
+
```
Comment (sriyen-msft) [inline certified-connectors/BttnV2/Readme.md:4]: @WesselBruijnes Please refer the template for readme.md file and add required sections.

File: certified-connectors/BttnV2/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+    "https"
+  ],
+  "consumes": [],
+  "produces": [],
+  "paths": {
+    "/api/company/1": {
+      "get": {
+        "responses": {
+          "default": {
+            "description": "default",
```
Comment (sriyen-msft) [inline certified-connectors/BttnV2/apiDefinition.swagger.json:25]: @WesselBruijnes Default responses are not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide valid expected response codes with schema definitions for all the operations in the swagger.

File: certified-connectors/BttnV2/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+            "schema": {}
```
Comment (sriyen-msft) [inline certified-connectors/BttnV2/apiDefinition.swagger.json:26]: @WesselBruijnes Empty response schema is not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide a valid response schema definition for all the operations in the swagger.

File: certified-connectors/BttnV2/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+          "items": {
+            "type": "object",
+            "properties": {
+              "event_id": {
+                "type": "string",
+                "description": "event_id"
+              },
+              "action_configuration_id": {
+                "type": "string",
+                "description": "action_configuration_id"
```
Comment (sriyen-msft) [inline certified-connectors/BttnV2/apiDefinition.swagger.json:50]: @WesselBruijnes All the description values in the swagger are expected to be free from grammatical and spell errors. Please provide valid description values for all the properties.

File: certified-connectors/BttnV2/apiProperties.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            "refreshUrl": {
+              "value": "https://sso.bt.tn/auth/realms/bttn/protocol/openid-connect/token"
+            }
+          }
+        }
+      }
+    },
+    "iconBrandColor": "#c51a1b",
+    "capabilities": [],
+    "publisher": "Wessel Bruijnes"
```
Comment (sriyen-msft) [inline certified-connectors/BttnV2/apiProperties.json:34]: @WesselBruijnes Can you please add the "stackOwner" property as well?

Comment (sriyen-msft) [inline certified-connectors/BttnV2/apiProperties.json:34]: @WesselBruijnes The publisher name can not be an individual name.  if you want to get this connector certified with an individual name, you need to move your connector to "independent-publisher connectors" folder.

File: certified-connectors/BttnV2/apiProperties.json [hunk 6]
```diff
@@ -0,0 +1,36 @@
+{
+  "properties": {
+    "connectionParameters": {
+      "token": {
+        "type": "oauthSetting",
+        "oAuthSettings": {
+          "identityProvider": "oauth2",
+          "clientId": "PowerAutomate",
```
Comment (sriyen-msft) [inline certified-connectors/BttnV2/apiProperties.json:8]: @WesselBruijnes Please remove th client id value and replace it with "[[TEST]]"

File: certified-connectors/BttnV2/apiDefinition.swagger.json (hunk 3 above)
Comment (sriyen-msft) [reply certified-connectors/BttnV2/apiDefinition.swagger.json:26]: Applies for all the responses in the swagger

File: certified-connectors/BttnV2/apiDefinition.swagger.json (hunk 2 above)
Comment (sriyen-msft) [reply certified-connectors/BttnV2/apiDefinition.swagger.json:25]: Applies for all the responses in the swagger

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

**Comment by Amjed-Ayoub** [comment]
Dear partner,

We apologize for the delay you might be experiencing while we are working on a high volume of connector requests. Rest assured that we are doing everything we can to address this volume. We are taking measures like improving our processes and hiring additional team members to guarantee resolving your connector requests in a timely manner. 

Please let us know if you have any questions in the meantime.

**Comment by sriyen-msft** [comment]
@WesselBruijnes I'm just checking in to see how things are going with updating your connector artifacts as per the given feedback. Please feel free to reach out with any questions you may have, Thanks.

**Comment by Amjed-Ayoub** [comment]
Hi @WesselBruijnes, I am following up with you to see how developing your connector is going?
Please let me know if you have any questions.
Thank you very much

**Comment by Amjed-Ayoub** [comment]
Hi @WesselBruijnes, I am following up with you to see how updating your connector is going?
Please let me know if you have any questions.
Thank you very much

**Comment by WesselBruijnes** [comment]
> Hi @WesselBruijnes, I am following up with you to see how updating your connector is going? Please let me know if you have any questions. Thank you very much

Hi, I am sorry for the long delay. We are starting up this process again. Could you possibly provide some up to date documentation? From the comments it seems like a lot required stuff is missing in the one available.

**Comment by Amjed-Ayoub** [comment]
> > Hi @WesselBruijnes, I am following up with you to see how updating your connector is going? Please let me know if you have any questions. Thank you very much
> 
> Hi, I am sorry for the long delay. We are starting up this process again. Could you possibly provide some up to date documentation? From the comments it seems like a lot required stuff is missing in the one available.

Hi @WesselBruijnes, Please see below for detailed information.
1.	Ensure that your connector complies with Microsoft requirements, as specified in [Meet submission requirements](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#step-2-meet-submission-requirements).
2.	Prepare a set of files called connector artifacts to Microsoft by following the instructions in [Prepare the connector artifacts](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#step-4-prepare-the-connector-artifacts).
3.	Ensure that your connector [contains valid metadata](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#step-3-add-metadata).
4.	Submit your connector by following the instructions in [Submit your connector to Microsoft for certification](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification).
a.	[Open source the connector files](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#open-source-the-connector). Our Certification Engineers will review within 1-2 weeks, and you will be notified in GitHub.
b.	[Upload from local files](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#upload-from-local-files) to ISV Studio (If you've been granted an exception from open sourcing). We will review your submission within 1-2 weeks.
c.	After the pull request is merged, you may [submit the files to ISV Studio](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio), please allow us 1-2 weeks to review your submission, and our Engineers will notify you in the “Activity Control” area in ISV Studio.
5.	[Create an environment in the Preview region](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#create-an-environment-in-the-preview-region). You will use this environment later to [test your connector](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#test-your-connector) after Microsoft is done with the functional verification for your connector. 
We expect all tests to be completed within 1-2 weeks.
6.	If your connector passes all tests, the deployment process begins, and it typically takes 5 to 6 weeks to deploy your connector to all of our regions.
For detailed steps and instructions, go to [Certification process](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission).

I hope that helps.

Please let me know if you have any questions.

Thank you very much

File: certified-connectors/BttnOne/Readme.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+* A Power Automate action created on the customer portal.
+
+## Authentication 
+Connect this connector to your Bttn account using Oauth. You are automatically forwarded to the Bttn login page.
+
+## Supported Operations
+The connector supports the following operations:
+* `getActionConfigurations`: Lists all your Power Automate actions configured on the Bttn portal.
+* `bttnWebhook`: Sets a webhook for your Power Automate flow on a configured Power Automate Action. This webhook wil be executed when the bttn ONE is pressed.
+
```
[the remark this answers is not in the document]
Comment (Amjed-Ayoub) [reply certified-connectors/BttnOne/Readme.md:21]: Please include all required fields in the readme

[the remark this answers is not in the document]
Comment (WesselBruijnes) [reply certified-connectors/BttnOne/Readme.md:26]: Hi Amjed, I have updated the readme to contain the required fields.

**Review by sriyen-msft — APPROVED** [review]

**Pull request merged**